### PR TITLE
fix: Upgrade to newer cert-manager

### LINKED ISSUE #5353: Cert Manager Compatibility before November 2019
### Summary 

Got this email from Let's Encrypt:

```
We've been working with Jetstack, the authors of cert-manager, on a
series of fixes to the client. Cert-manager sometimes falls into a
traffic pattern where it sends really excessive traffic to Let's
Encrypt's servers, continuously. To mitigate this, we plan to start
blocking all traffic from cert-manager versions less than 0.8.0 (the
current semver minor release), as of November 1, 2019. Please upgrade
all of your cert-manager instances before then.

We're sending this email because this is the contact address of your
cert-manager instance at:

REMOVED

Version 0.8.0 is much better but we still observe excessive traffic in
some cases. We're working with Jetstack to improve these cases. As new
versions of cert-manager are released, we will add the non-current
versions to our block list after 3 months. We strongly encourage
cert-manager users to stay up-to-date with new versions.

Also, there is an opportunity to help both Jetstack and Let's Encrypt.
Once you've upgraded, please check the logs for your cert-manager
instances from time to time. Are they making excessive requests to Let's
Encrypt (more than, say, 10 per day over multiple days)? If so, please
share details at https://github.com/jetstack/cert-manager/issues/1948

Thanks,
Let's Encrypt Team
```

I believe JX is currently using 0.6 and this suggests using 0.8 at a minimum, but also warns that future versions might be blocked.  We currently manage the our install of `cert-manager`, but we want to make sure JX is compatible with future versions of it.

You can review releases here:
https://github.com/jetstack/cert-manager/releases

You can see upgrade steps/actions for each release under here:
http://docs.cert-manager.io/en/latest/tasks/upgrading/index.html

### Steps to reproduce the behavior

N/A

### Expected behavior

JX is compatible with latest versions of `cert-manger` (at _least_ 0.8).

### Actual behavior

JX is using `cert-manger` 0.6.

### Jx version

N/A

### Jenkins type

- [ ] Serverless Jenkins X Pipelines (Tekton + Prow)
- [X] Classic Jenkins

### Kubernetes cluster

AWS via KOPS

### Operating system / Environment

N/A

### PR DESCRIPTION
**Pull request by abayer**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

Trying to address #5353 - the initial attempt to do so in #5464 failed, for a couple reasons, and was reverted. So let's try again and see exactly what's wrong at this point.

#### Special notes for the reviewer(s)

/assign @jstrachan 

#### Which issue this PR fixes

fixes #5353

### CODE REVIEW

**Comment by abayer** [comment]
Failing test on tekton context:
```
upgrade ingress
 Given valid parameters when running upgrade ingress
  creates the ingress resource and fetch a TLS certificate for services from specified namespace

  /tmp/jenkins-x/bdd-jx/test/suite/ingress/jx_upgrade_ingress.go:302
      RUNNER:About to execute jx upgrade ingress -b --skip-resources-update --namespaces=test-upgrade-ingress --config-namespace=test-upgrade-ingress --wait-for-certs in /tmp/bdd-906697468 with timeout 5m0s expecting exit code 0
Looking for "cert-manager" deployment in namespace "cert-manager"...
Installing cert-manager...
Installing CRDs from "https://raw.githubusercontent.com/jetstack/cert-manager/v0.9.1/deploy/manifests/00-crds.yaml"...
customresourcedefinition.apiextensions.k8s.io/certificates.certmanager.k8s.io created
customresourcedefinition.apiextensions.k8s.io/certificaterequests.certmanager.k8s.io created
customresourcedefinition.apiextensions.k8s.io/challenges.certmanager.k8s.io created
customresourcedefinition.apiextensions.k8s.io/clusterissuers.certmanager.k8s.io created
customresourcedefinition.apiextensions.k8s.io/issuers.certmanager.k8s.io created
customresourcedefinition.apiextensions.k8s.io/orders.certmanager.k8s.io created
Ensuring helm repo "jetstack" at "https://charts.jetstack.io" for cert-manager chart is configured
Installing the chart "jetstack/cert-manager" in namespace "cert-manager"...
Namespace cert-manager created

Waiting for CertManager deployment to be ready, this can take a few minutes
Expecting certificates: [test-upgrade-ingress/tls-test-svc]
Certificate issuer letsencrypt-staging does not exist. Creating...
error: creating the ingress rules: creating the cert-manager issuer test-upgrade-ingress/letsencrypt-staging: creating cert-manager issuer "letsencrypt-staging": Issuer.certmanager.k8s.io "letsencrypt-staging" is invalid: []: Invalid value: map[string]interface {}{"apiVersion":"certmanager.k8s.io/v1alpha1", "metadata":map[string]interface {}{"name":"letsencrypt-staging", "creationTimestamp":"2019-09-23T17:13:04Z", "namespace":"test-upgrade-ingress", "generation":1, "uid":"678b2e7d-de25-11e9-8a97-42010a84005b"}, "spec":map[string]interface {}{"acme":map[string]interface {}{"email":"test@jenkinsx.io", "server":"https://acme-staging-v02.api.letsencrypt.org/directory", "privateKeySecretRef":map[string]interface {}{"name":"letsencrypt-staging", "key":""}, "http01":map[string]interface {}{}}}, "status":map[string]interface {}{"conditions":interface {}(nil)}, "kind":"Issuer"}: validation failure list:
status.conditions in body must be of type array: "null"
      RUNNER:Execution completed with exit code 1
      FAILED got unexpected error:

expected exit code 0 but got 1 whilst running command jx upgrade ingress -b --skip-resources-update --namespaces=test-upgrade-ingress --config-namespace=test-upgrade-ingress --wait-for-certs
```

**Comment by abayer** [comment]
Same error as reported on https://github.com/jenkins-x/jx/issues/4550. @jstrachan - I think cert-manager 0.9.1 may need a newer version of Kubernetes than we allow as a minimum?

**Comment by abayer** [comment]
Another possibility is that the `Issuer` we're creating is being serialized with our cert-manager 0.5.2 dependency, which doesn't have `omitempty` on `status.conditions`. I tried bumping our build dependency to cert-manager 0.9.1 and...well, that required things like bumping our k8s dependencies (including code gen) to 1.15, which seemed risky to me.

**Comment by abayer** [comment]
Well, manually adding an empty `Conditions` array (and a `Status`) did the trick. Color me surprised but happy.

File: pkg/cmd/opts/cert_manager.go [hunk 1]
```diff
@@ -37,6 +37,14 @@ func (o *CommonOptions) EnsureCertManager() error {
 			}
 			log.Logger().Info(output)
 
+			log.Logger().Infof("Ensuring helm repo %q at %q for cert-manager chart is configured", pki.CertManagerChartOwner,
```
Comment (hferentschik) [inline pkg/cmd/opts/cert_manager.go]: Interesting, never used %q before

File: pkg/kube/pki/cert-manager.go [hunk 2]
```diff
@@ -17,10 +17,14 @@ const (
...
+	// CertManagerChartOwner is the owner of the cert-manager chart repo
+	CertManagerChartOwner = "jetstack"
+	// CertManagerChartURL is the URL for the repo containing the cert-manager chart
+	CertManagerChartURL = "https://charts.jetstack.io"
 	// CertManagerChart name of the cert-manager chart
-	CertManagerChart = "stable/cert-manager"
+	CertManagerChart = "jetstack/cert-manager"
 	// CertManagerCRDsFile files which contains the cert-manager CRDs
-	CertManagerCRDsFile = "https://raw.githubusercontent.com/jetstack/cert-manager/release-0.6/deploy/manifests/00-crds.yaml"
+	CertManagerCRDsFile = "https://raw.githubusercontent.com/jetstack/cert-manager/v0.9.1/deploy/manifests/00-crds.yaml"
```
Comment (hferentschik) [inline pkg/kube/pki/cert-manager.go]: ooi, is this not also something which should come in some form out of the version stream? Seems odd to have a hard coded version here, right?

**Pull request merged**